sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -204,8 +204,8 @@ def report_result(result_code, batch)
             end
           end
 
-          def report_dropped_spans(count, reason:)
-            @metrics_reporter.add_to_counter('otel.bsp.dropped_spans', increment: count, labels: { 'reason' => reason })
+          def report_dropped_spans(count, labels = {})
```

Comment:
Although I changed the method signature, we do not need to change other call sites for `report_dropped_spans`  since they were using the kwarg `reason: 'foo'` which we can also interpret as a hash.

Reply:
🤔 I'd rather we used named arguments consistently, and Strings for keys and values in the labels. I.e. I'd prefer:
```ruby
def report_dropped_spans(count, labels: nil) ... end

report_dropped_spans(n, labels: { 'reason' => 'buffer-full', 'code.function' => 'force_flush' })
```
or:
```ruby
def report_dropped_spans(count, reason:, function: nil)
  @metrics_reporter.add_to_counter('otel.bsp.dropped_spans', increment: count, labels: { 'reason' => reason, 'code.function' => function }.compact)
end

report_dropped_spans(n, reason: 'buffer-full', function: 'force_flush')
```

I prefer the latter, since it is cleaner for callers.

Reply:
Yes, let's do the latter

Reply:
dfab14e06e961a925be24be69d4ae91d0d463814